src/pages/data-entry/data-entry.jsx

```diff
@@ -1,127 +1,214 @@
-import React, { Component } from "react";
+import React, { Component } from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import { useMutation } from '@apollo/react-hooks'
+import { gql } from 'apollo-boost'
+
+const units = gql`
+  {
+    getAllAvailableUnits
+    }
+  }
+`
+
+const insertMutation = gql`
+  mutation InsertMutation(
+    $unit: String!
+    $value: Float!
+    $latitude: Float!
+    $longitude: Float!
+  ) {
+    insertValue(
+      unit: $unit
+      value: $value
+      latitude: $latitude
+      longitude: $longitude
+    )
+  }
+`
+
+function DataFunction () {
+  const { loading, error, data } = useQuery(units)
+
+  if (loading) return 'Loading...'
+  if (error) return `Error! ${error.message}`
+
+  return data.getAllAvailableUnits
+}
 
 export class DataEntryForm extends Component {
-  constructor(props) {
-    super(props);
+  constructor (props) {
+    super(props)
 
     this.state = {
-      age: "",
-      sex: "",
-      weight: "",
-      wunits: "",
-      height: "",
-      hunits: "",
+      age: '',
+      sex: '',
+      weight: '',
+      // For all of our units, we have hypothetical arrays called WeightUnitsArray and HeightUnitsArray
+      // These are 2-D arrays with the first column dedicated to the unit name, and the second part dedicated to its ratio in comparison to 1 kg or 1 m.
+      // These Arrays should be initialized with the any common units (kg, lb, m, and ft).
+      wunits: '',
+      height: '',
+      hunits: '',
       errorsex: false,
       errorhunits: false,
       errorwunits: false
-    };
-
-    this.addweight = this.addweight.bind(this);
-    this.addweightUnits = this.addweightUnits.bind(this);
-    this.addheight = this.addheight.bind(this);
-    this.addheightUnits = this.addheightUnits.bind(this);
-    this.addage = this.addage.bind(this);
-    this.setsex = this.setsex.bind(this);
-    this.enterForm = this.enterForm.bind(this);
+    }
+
+    this.addweight = this.addweight.bind(this)
+    this.addweightUnits = this.addweightUnits.bind(this)
+    this.addheight = this.addheight.bind(this)
+    this.addheightUnits = this.addheightUnits.bind(this)
+    this.addage = this.addage.bind(this)
+    this.setsex = this.setsex.bind(this)
+    this.enterForm = this.enterForm.bind(this)
   }
 
-  addweight(w) {
-    this.setState({ ...this.state, ...{ weight: w.target.value } });
+  addweight (w) {
+    this.setState({ ...this.state, ...{ weight: w.target.value } })
   }
-  addweightUnits(unitw) {
-    if (unitw.target.value !== "lb" && unitw.target.value !== "kg") {
-      this.setState({ ...this.state, ...{ errorwunits: true } });
-    } else {
-      this.setState({ ...this.state, ...{ wunits: unitw.target.value } });
-      this.setState({ ...this.state, ...{ errorwunits: false } });
+  addweightUnits (unitw) {
+    let WeightUnitsArray = DataFunction()
+
+    unitw.target.value.toLowerCase()

```

Comment:
What does this piece of code do, you are not assigning the value of the lowercase string to anything.